File: EssentialToolsCs/EssentialToolsCs/Models/DefaultDiscountHelper.cs
namespace EssentialToolsCs.Models
{
    public class DefaultDiscountHelper : IDiscountHelper
    {
        // property injection
        // public decimal DiscountSize { get; set; }

        // constructor injection
        private readonly decimal discountSize;

        public DefaultDiscountHelper(decimal discountParam)
        {
            discountSize = discountParam;
        }

        public decimal ApplyDiscount(decimal totalParam)
        {
            // return (totalParam - (10m / 100m * totalParam));
            // property injection
            // return (totalParam - (DiscountSize / 100m * totalParam));
            // constructor injection
            return (totalParam - (discountSize / 100m * totalParam));
        }
    }
}

File: EssentialToolsVb/EssentialToolsVb/Models/DefaultDiscountHelper.vb
Imports EssentialToolsVb

Public Class DefaultDiscountHelper
    Implements IDiscountHelper

    ' property injection
    ' Public Property DiscountSize() As Decimal

    'constructor injection
    ReadOnly discountSize As Decimal

    Public Sub New(discountParam As Decimal)
        discountSize = discountParam
    End Sub

    Public Function ApplyDiscount(totalParam As Decimal) As Decimal Implements IDiscountHelper.ApplyDiscount
        ' Return (totalParam - (10D / 100D * totalParam))
        ' property injection
        ' Return (totalParam - (DiscountSize / 100D * totalParam))
        'constructor injection
        Return (totalParam - (discountSize / 100D * totalParam))
    End Function
End Class